zebra-consensus/src/transaction.rs

```diff
@@ -472,12 +455,11 @@ where
                 // description while adding the resulting future to
                 // our collection of async checks that (at a
                 // minimum) must pass for the transaction to verify.
-                let rsp = redjubjub_verifier
-                    .ready_and()
-                    .await?
-                    .call((spend.rk, spend.spend_auth_sig, &shielded_sighash).into());
-
-                async_checks.push(rsp.boxed());
+                async_checks.push(
+                    primitives::redjubjub::VERIFIER
+                        .clone()
+                        .oneshot((spend.rk, spend.spend_auth_sig, shielded_sighash).into()),
+                );
```

Comment:
So this is going to clone the redjubjub verifier service for every signature we want to verify?

Reply:
Similarly, I'm not too concerned about cloning a few pointers, given that the underlying `BatchControl` and semaphore are shared between all the clones.

Happy to revisit this when we have more time though.

Reply:
Yeah not a blocker, just want to make sure we're not accidentally bloating up our heap cyclically depending on how many signatures we verify